### DISCUSSION TITLE
StreamSystem CreateStream timeout

### DISCUSSION THREAD

**Discussion by kallencst** [rabbitmq/rabbitmq-stream-dotnet-client#207] [Q&A]
I am attempting to create a WPF application that contains the sample code you provided under "Getting Started". I am able to create a StreamSystem using the config as shown, with UserName and Password specified as "guest" and VirtualHost as "/". But when CreateStream is called, after about 10 seconds, a System.Timeout exception occurs, with this stack trace: 

<details>

```
"   at RabbitMQ.Stream.Client.ManualResetValueTaskSource`1.System.Threading.Tasks.Sources.IValueTaskSource<T>.GetResult(Int16 token)\r\n   at RabbitMQ.Stream.Client.Client.<Request>d__54`2.MoveNext()\r\n   at RabbitMQ.Stream.Client.Client.<Request>d__54`2.MoveNext()\r\n   at RabbitMQ.Stream.Client.Client.<CreateStream>d__70.MoveNext()\r\n   at RabbitMQ.Stream.Client.StreamSystem.<CreateStream>d__15.MoveNext()\r\n   at RabbitMQStreamsProducer.MainWindow.<Window_Loaded>d__6.MoveNext() in C:\\Users\\kent.allen\\source\\repos\\RabbitMQStreamsProducer\\RabbitMQStreamsProducer\\MainWindow.xaml.cs:line 48". 
```
</details>

I am able to see the RabbitMQ service log file, it is very verbose but contains this:

<details>

```
2023-01-09 14:27:04.670000-06:00 [error] <0.2430.0> ** Reason for termination ==
2023-01-09 14:27:04.670000-06:00 [error] <0.2430.0> ** {{case_clause,{protocol_error,internal_error,
2023-01-09 14:27:04.670000-06:00 [error] <0.2430.0>                                  "Cannot declare a queue '~s' on node '~s': ~255p",
2023-01-09 14:27:04.670000-06:00 [error] <0.2430.0>                                  ["queue 'my_first_stream' in vhost '/'",
2023-01-09 14:27:04.670000-06:00 [error] <0.2430.0>                                   'rabbit@LKM-LCS009704',
2023-01-09 14:27:04.670000-06:00 [error] <0.2430.0>                                   {error,coordinator_unavailable}]}},
```
</details>

Here is the code I am using:

<details><summary>C# code</summary>

```csharp
using System;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using RabbitMQ.Stream.Client;
using RabbitMQ.Stream.Client.Reliable;

namespace RabbitMQStreamsProducer
{
    public partial class MainWindow : Window
    {
        private StreamSystemConfig _config = new StreamSystemConfig
        {
            UserName = "guest",
            Password = "guest",
            VirtualHost = "/"
        };

        const string _stream = "my_first_stream";

        private StreamSystem _system;

        public MainWindow()
        {
            InitializeComponent();
        }

        private async void Window_Loaded(object sender, RoutedEventArgs e)
        {
            _system = await StreamSystem.Create(_config);

            await _system.CreateStream(new StreamSpec(_stream)
            {
                MaxLengthBytes = 200000,
            }); // Timeout occurs here

            // Producer code follows, omitted
        }
    }
```

Thanks for any advice.

**Comment by lukebakken** [comment]
Have you enabled the necessary plugin in RabbitMQ?

https://www.rabbitmq.com/stream.html#enabling-plugin

**Reply by kallencst** [reply]
I believe so, I entered this command: rabbitmq-plugins enable rabbitmq_stream. This apparently succeeded. Are any other plugins needed?

**Reply by lukebakken** [reply]
Could you run these command and capture the commands as well as the output? The second command will produce a lot of output so please redirect it to a file and attach the file.

```
rabbitmq-plugins list
rabbitmqctl report
```

**Comment by kallencst** [comment]
I was able to capture the list of plugins (attached). 
rabbitmq-plugins list > list-of-plugins.txt
[list-of-plugins.txt](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/files/10377111/list-of-plugins.txt)

The "rabbitmqctl report" command failed. Here is what appears in the cmd window:

<details>

```
rabbitmqctl report > ctl-report.txt
Error: unable to perform an operation on node 'rabbit@LKM-LCS009704'. Please see diagnostics information and suggestions below.

Most common reasons for this are:

 * Target node is unreachable (e.g. due to hostname resolution, TCP connection or firewall issues)
 * CLI tool fails to authenticate with the server (e.g. due to CLI tool's Erlang cookie not matching that of the server)
 * Target node is not running

In addition to the diagnostics info below:

 * See the CLI, clustering and networking guides on https://rabbitmq.com/documentation.html to learn more
 * Consult server logs on node rabbit@LKM-LCS009704
 * If target node is configured to use long node names, don't forget to use --longnames with CLI tools

DIAGNOSTICS
===========

attempted to contact: ['rabbit@LKM-LCS009704']

rabbit@LKM-LCS009704:
  * connected to epmd (port 4369) on LKM-LCS009704
  * epmd reports node 'rabbit' uses port 25672 for inter-node and CLI tool traffic
  * TCP connection succeeded but Erlang distribution failed
  * suggestion: check if the Erlang cookie is identical for all server nodes and CLI tools
  * suggestion: check if all server nodes and CLI tools use consistent hostnames when addressing each other
  * suggestion: check if inter-node connections may be configured to use TLS. If so, all nodes and CLI tools must do that
   * suggestion: see the CLI, clustering and networking guides on https://rabbitmq.com/documentation.html to learn more


Current node details:
 * node name: 'rabbitmqcli-305-rabbit@LKM-LCS009704'
 * effective user's home directory: c:/Users/kent.allen
 * Erlang cookie hash: T+k/VcgS0A6lKoBLSEGlHw==
```
</details>

**Reply by lukebakken** [reply]
Please make a note of how I've edited your comments to use `<details></details>` so that it's much easier to follow the discussion.

You need to follow the steps here to ensure your user is using the same cookie as RabbitMQ:
https://www.rabbitmq.com/install-windows-manual.html#erlang-cookie

Basically, copy the `C:\Windows\system32\config\systemprofile\.erlang.cookie` file to your user's home directory.

**Comment by kallencst** [comment]
Thanks, I did try copying that file but it did not help. Does anything have to be restarted/reset or whatever after copying the file?

**Reply by lukebakken** [reply]
Please get a transcript of the commands you run and their output. Without it I can't verify what you've done.

**Comment by kallencst** [comment]
I had another person try it on a different machine, and everything works fine for them. It seems like something must be wrong with my installation. I will try starting over from scratch.

**Reply by Gsantomaggio** [reply]
The problem is more server-side.  

>I will try starting over from scratch.

let us know

**Comment by kallencst** [comment]
It works now. Thanks for helping! When I re-installed, I used the Windows installer and did not use Chocolatey, not sure if that is what made the difference.

**Reply by lukebakken** [reply]
Thanks for letting us know. That's odd, but glad you got it working!

**Comment by Gsantomaggio** [comment]
will close due of inactivity
